gke-policies/policy/vpc_native_cluster.rego

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# METADATA
+# title: GKE VPC-native cluster
+# description: GKE cluster nodepool should be VPC-native as per our best-practices
+# custom:
+#   group: Management
+package gke.policy.vpc_native_cluster
+
+default valid = false
+
+valid {
+  count(violation) == 0
+}
+
+violation[msg] {
+  not input.node_pools[pool].network_config.pod_range
```

Comment:
Isn't it enough to check this on a cluster level ?

Checking it on a node pool will do the job, but I think that when cluster is VPC native, all node pools need to have pod range (but pls correct me if I'm wrong).

Consider maybe checking one of those:
* for VPC native cluster there is `input.ip_allocation_policy.use_ip_aliases`
* for route base cluster there is `input.ip_allocation_policy.use_routes`

Reply:
Please add some tests as well :-)